### PR TITLE
DAOS-13393 gurt: Add thread local setting to fault injection.

### PR DESCRIPTION
**Pull request by ashleypittman**
Allow faults to be enabled/disabled on a per thread basis.
This paves the way for running better fault inection testing
on asyncronous operations where they are submitted and
completed on different threads.

Re-format the fault inject code and remove a header file.

Required-githooks: true
Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/gurt/fault_inject.c [hunk 1]
```diff
@@ -82,63 +79,117 @@ fa_op_rec_decref(struct d_hash_table *htab, d_list_t *rlink)
...
 
-	rlink = d_hash_rec_find_insert(&d_fi_gdata.dfg_fa_table, &fault_id,
-				sizeof(fault_id), &new_rec->fa_link);
+	rlink = d_hash_rec_find_insert(&d_fi_gdata.dfg_fa_table, &fault_id, sizeof(fault_id),
+				       &new_rec->fa_link);
 	if (rlink == &new_rec->fa_link) {
 		fault_attr = &new_rec->fa_attr;
-		rc = D_SPIN_INIT(&fault_attr->fa_lock,
-				 PTHREAD_PROCESS_PRIVATE);
+		rc         = D_SPIN_INIT(&fault_attr->fa_lock, PTHREAD_PROCESS_PRIVATE);
```
Comment (frostedcmos) [inline src/gurt/fault_inject.c:185]: strange formatting here:)

File: src/gurt/fault_inject.c [hunk 2]
```diff
@@ -82,63 +79,117 @@ fa_op_rec_decref(struct d_hash_table *htab, d_list_t *rlink)
...
+static uint32_t            d_fault_inject_seed;
+static struct d_fi_gdata_t d_fi_gdata;
+static pthread_once_t      d_fi_gdata_init_once = PTHREAD_ONCE_INIT;
+
+/* Check for and set thread specific enabled  flag.  This is a single function that returns two
+ * values so it can access a single _thread local.
+ *
+ * 0 - no value set;
+ * 1 - disabled;
+ * >2 - enabled;
```
Comment (frostedcmos) [inline src/gurt/fault_inject.c:108]: would be more readable to add defines for those, e.g. FAULTS_NOT_SET, FAULTS_DISABLED, FAULTS_ENABLED or some such.

**Review by frostedcmos — APPROVED** [review]
2 minor comments inline

File: src/gurt/fault_inject.c (hunk 1 above)
Comment (ashleypittman) [reply src/gurt/fault_inject.c:185]: You're right, it's aligning assignments so it's the same as the line above.  I'll add a blank line and it'll put it back where you expect.

File: src/gurt/fault_inject.c (hunk 2 above)
Comment (ashleypittman) [reply src/gurt/fault_inject.c:108]: If this wasn't a single static function in a header file I'd agree.

**Review by daltonbohning — APPROVED** [review]

**Review by frostedcmos — APPROVED** [review]

**Pull request merged**